RecoPixelVertexing/PixelTriplets/plugins/CAHitTripletGenerator.cc

```diff
@@ -0,0 +1,244 @@
+#include <unordered_map>
+
+#include "CAHitTripletGenerator.h"
+
+#include "RecoPixelVertexing/PixelTriplets/interface/ThirdHitPredictionFromCircle.h"
+
+#include "RecoPixelVertexing/PixelTriplets/interface/HitTripletGenerator.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "DataFormats/Common/interface/Handle.h"
+
+#include "TrackingTools/DetLayers/interface/BarrelDetLayer.h"
+
+#include "CellularAutomaton.h"
+
+#include "CommonTools/Utils/interface/DynArray.h"
+
+#include "FWCore/Utilities/interface/isFinite.h"
+
+namespace
+{
+
+template<typename T>
+T sqr(T x)
+{
+	return x * x;
+}
+}
+
+using namespace std;
+using namespace ctfseeding;
+
+CAHitTripletGenerator::CAHitTripletGenerator(const edm::ParameterSet& cfg,
+		edm::ConsumesCollector& iC) :
+		theSeedingLayerToken(
+				iC.consumes < SeedingLayerSetsHits
+						> (cfg.getParameter < edm::InputTag > ("SeedingLayers"))), extraHitRPhitolerance(
+				cfg.getParameter<double>("extraHitRPhitolerance")), //extra window in ThirdHitPredictionFromCircle range (divide by R to get phi)
+		maxChi2(cfg.getParameter < edm::ParameterSet > ("maxChi2")), useBendingCorrection(
+				cfg.getParameter<bool>("useBendingCorrection")), CAThetaCut(
+				cfg.getParameter<double>("CAThetaCut")), CAPhiCut(
+				cfg.getParameter<double>("CAPhiCut")), CAHardPtCut(
+				cfg.getParameter<double>("CAHardPtCut"))
```

Comment:
this code would be easier to parse if single parameter initialization is done on one line and not split and mixed with others